payment-request/interfaces.https.html

```diff
@@ -5,136 +5,27 @@
-  required DOMString supportedMethods;
-  PaymentItem total;
-  sequence<PaymentItem> additionalDisplayItems;
-  object data;
-};
-enum PaymentShippingType {
-  "shipping",
-  "delivery",
-  "pickup"
-};
-dictionary PaymentOptions {
-  boolean requestPayerName = false;
-  boolean requestPayerEmail = false;
-  boolean requestPayerPhone = false;
-  boolean requestShipping = false;
-  PaymentShippingType shippingType = "shipping";
-};
-dictionary PaymentItem {
-  required DOMString label;
-  required PaymentCurrencyAmount amount;
-  boolean pending = false;
-};
-[SecureContext]
-interface PaymentAddress {
-  [Default] object toJSON();
-  readonly attribute DOMString country;
-  readonly attribute FrozenArray<DOMString> addressLine;
-  readonly attribute DOMString region;
-  readonly attribute DOMString city;
-  readonly attribute DOMString dependentLocality;
-  readonly attribute DOMString postalCode;
-  readonly attribute DOMString sortingCode;
-  readonly attribute DOMString languageCode;
-  readonly attribute DOMString organization;
-  readonly attribute DOMString recipient;
-  readonly attribute DOMString phone;
-};
-dictionary PaymentShippingOption {
-  required DOMString id;
-  required DOMString label;
-  required PaymentCurrencyAmount amount;
-  boolean selected = false;
-};
-enum PaymentComplete {
-  "fail",
-  "success",
-  "unknown"
-};
-[SecureContext]
-interface PaymentResponse {
-  [Default] object toJSON();
-
-  readonly attribute DOMString requestId;
-  readonly attribute DOMString methodName;
-  readonly attribute object details;
-  readonly attribute PaymentAddress? shippingAddress;
-  readonly attribute DOMString? shippingOption;
-  readonly attribute DOMString? payerName;
-  readonly attribute DOMString? payerEmail;
-  readonly attribute DOMString? payerPhone;
-
-  Promise<void> complete(optional PaymentComplete result = "unknown");
-};
-[Constructor(DOMString type, optional PaymentRequestUpdateEventInit eventInitDict),
-  SecureContext]
-interface PaymentRequestUpdateEvent : Event {
-  void updateWith(Promise<PaymentDetailsUpdate> detailsPromise);
-};
-dictionary PaymentRequestUpdateEventInit : EventInit {
-};
-</script>
-
 <script>
-"use strict";
-var idlArray = new IdlArray();
-[].forEach.call(document.querySelectorAll("script[type=text\\/plain]"), function(node) {
-  if (node.className == "untested") {
-    idlArray.add_untested_idls(node.textContent);
-  } else {
-    idlArray.add_idls(node.textContent);
+promise_test(async () => {
+  const idlURLs = ["/interfaces/dom.idl", "/interfaces/payment-request.idl"];
+  const idlArray = new IdlArray();
+  for(const url of idlURLs){
+    const idlText = await fetch(url).then(r => r.text());
+    idlArray.add_idls(idlText);
   }
-});
-idlArray.add_objects({
-  PaymentRequest: ["new PaymentRequest([{supportedMethods: 'foo'}], {total: {label: 'bar', amount: {currency: 'BAZ', value: '0'}}})"]
-});
-idlArray.test();
+  // typedef EventHandler from HTML
```

Comment:
Filed a bug on DOM spec regarding having to do this:
https://github.com/whatwg/dom/issues/509

Let's see what @annevk thinks.

Reply:
Anne suggested maybe splitting out EventHandler into it's own IDL file. That might make sense given how many specs are relying on it (but where it wouldn't make sense to import all of HTML). 

See: 

```
git grep "interface EventHandler"
```

Reply:
(can do this as a separate bug, if we agree to that)

Reply:
Proposal: https://github.com/w3c/web-platform-tests/pull/7331